Extension/package.json

```diff
@@ -5092,27 +5092,27 @@
       "editor/context": [
         {
           "command": "C_Cpp.SwitchHeaderSource",
-          "when": "config.C_Cpp.intelliSenseEngine != 'Disabled' && editorLangId == 'c' || config.C_Cpp.intelliSenseEngine != 'Disabled' && editorLangId == 'cpp' || config.C_Cpp.intelliSenseEngine != 'Disabled' && editorLangId == 'cuda-cpp'",
+          "when": "(config.C_Cpp.intelliSenseEngine != 'disabled' || config.C_Cpp.intelliSenseEngine != 'Disabled') && editorLangId == 'c' || (config.C_Cpp.intelliSenseEngine != 'disabled' || config.C_Cpp.intelliSenseEngine != 'Disabled') && editorLangId == 'cpp' || (config.C_Cpp.intelliSenseEngine != 'disabled' || config.C_Cpp.intelliSenseEngine != 'Disabled') && editorLangId == 'cuda-cpp'",
```

Comment:
Are you sure this works? I recall the last time we weren't able to get `(a || b) && (c || d)` to be evaluated correctly by VS Code, i.e. it only worked with `a || b && c || d` style tests.

Reply:
It's an open issue: [Add support for parenthesis in "when" conditions #91473](https://github.com/microsoft/vscode/issues/91473)

https://code.visualstudio.com/api/references/when-clause-contexts

Reply:
I didn't realize there was an issue with that, I can just remove the parenthesis.

Reply:
I think it can be re-written, but you can't just remove the parens, because the && gets precedence.

Reply:
What I commented on earlier is that:
```
(config.C_Cpp.intelliSenseEngine != 'disabled' || config.C_Cpp.intelliSenseEngine != 'Disabled')
```
should actually be
```
config.C_Cpp.intelliSenseEngine != 'disabled' && config.C_Cpp.intelliSenseEngine != 'Disabled'
```
So you don't need parentheses if the `||` separates all the `&&` conditions

Reply:
Ah, yeah, `config.C_Cpp.intelliSenseEngine != 'disabled' || config.C_Cpp.intelliSenseEngine != 'Disabled'` would always evaluate to true.